android/shared_config.gradle

```diff
@@ -0,0 +1,28 @@
+import groovy.json.JsonSlurper
+
+/*
+ * This function returns the relevant React Native version. Before doing so,
+ * it performs a check to insure that the version for React Native in package.json
+ * has not changed. It does this so that anytime the version of React Native in
+ * package.json is changed, this method will fail the build until this method is also
+ * updated to return the new React Native version applicable for Android.
+ */
+ext.getRnVersion = { String pathToPackageJson, String section ->
+    def actualPackageJsonRnVersion = parseRnVersion(pathToPackageJson, section)
+    def expectedPackageJsonRnVersion = 'jtreanor/react-native#v0.60.0-patched'
+    if (actualPackageJsonRnVersion == expectedPackageJsonRnVersion) {
+        return '0.60.0-patched'
```

Comment:
I think it feels weird to have to look in this non-standard file, inside the demo app android folder to check for what version the expected one is. The native parts of the demo app don't play any role when building the bridge module, for example, and that's what a WPAndroid develop would probably look first to understand the dependencies.

In a sense, what I'm still trying to understant is whether the `gutenberg-mobile/package.json` can or cannot be the (single) source of truth for which RN version should be picked up 🤔.

Reply:
>I think it feels weird to have to look in this non-standard file, inside the demo app android folder to check for what version the expected one is. The native parts of the demo app don't play any role when building the bridge module, for example, and that's what a WPAndroid develop would probably look first to understand the dependencies.

My thinking was that it was an Android-specific file, so I put it in `android/`. I see your point though, perhaps this would be better (a) down one level at the root; or (b) in `react-native-gutenberg-bridge/android`. WDYT?

>In a sense, what I'm still trying to understand is whether the gutenberg-mobile/package.json can or cannot be the (single) source of truth for which RN version should be picked up 🤔.

In principle, yes. The reason I went with this approach was that _before we updated to RN 0.61.5_, the version numbers between the package.json and Android did not align well (package.json was vXXX, whereas Android was just XXX). This meant that to parse the version from the package.json would have involved some one-off fragile arbitrary parsing logic (i.e., check if there's a `v` after `#` and if there is drop the `v`) that I didn't want to use and then have lingering there forever. I put the check that throws an exception if the package.json version changed with the idea that it would force us to update this function to something more reasonable when we changed the version in package.json.

So... we've updated the RN version, which means I should probably get on updating this function 😄, and it does look like we should be able to now directly parse the version from the package.json file easily.

Reply:
Made the changes. This is now actually back to being more similar to what you had set up. As I see it, the two main differences are:

1. We are storing the function to parse the react-native version in a gradle file inside the repository instead of in a github gist; and
2. There are jest tests verifying that the react-native version between the different `package.json` files are consistent.